### PR TITLE
add GIT_BRANCH_NAME environment variable

### PR DESCRIPTION
**Pull request by sps**
see:
http://stackoverflow.com/questions/20541609/how-to-receive-local-git-branch-name-with-jenkins-git-plugin
http://stackoverflow.com/questions/14985563/how-to-retrieve-the-git-branch-name-that-was-built-by-jenkins-when-using-inverse

### CODE REVIEW

File: src/main/java/hudson/plugins/git/GitSCM.java [hunk 1]
```diff
@@ -1175,6 +1179,13 @@ private String getBranchName(Branch branch)
         return name;
     }
 
+
+    private String getShortBranchName(Branch branch) {
+        String name = firstNonNull(branch.getName(), "");
+        Iterable<String> parts = Splitter.on("/").omitEmptyStrings().split(name);
+        return Iterables.getLast(parts);
+    }
+
```
Comment (MarkEWaite) [inline src/main/java/hudson/plugins/git/GitSCM.java]: Is that really what you want?  I use a branch named develop/pre_8.3 for the work on a branch that will eventually be the 8.3 release of the product.  It is coming from a remote named origin.  I believe that your algorithm will return "pre_8.3" when my branch name is "develop/pre_8.3".

Comment (sps) [inline src/main/java/hudson/plugins/git/GitSCM.java]: Hey Mark, 

That was my original intent (i even wrote a unit test to cover that) but it's a fair point - some folks name their branches with "/" in them.

While I think this satisfies the majority of branch naming conventions, setting this value as GIT_BRANCH_NAME would be incorrect. Perhaps GIT_BRANCH_SHORT_NAME ?
Any suggestions?

Comment (KostyaSha) [inline src/main/java/hudson/plugins/git/GitSCM.java]: "develop/per_8.3" is a short name, long branch name just has repo identifier.

**Comment by sps** [comment]
Updated to omit the repo identifier only

**Comment by sps** [comment]
After doing some integration testing locally, I am seeing an issue. I am going to close this and resubmit when I have a better fix.

**Comment by AnneTheAgile** [comment]
This other ticket, also referenced in the SO post, above, was merged and enables direct checkout to a local branch instead of a detached head.
https://github.com/jenkinsci/git-plugin/pull/381

**Pull request closed without merging**